src/source.cpp

```diff
@@ -261,6 +269,50 @@ void initialize_source()
 
     // Close file
     file_close(file_id);
+  } else if ( settings::path_source_library != "" ) {
+    // Get the source from a library object 
+
+    std::stringstream msg;
+    msg << "Sampling from library source " << settings::path_source << "...";
+    write_message(msg, 6);
+
+    // Open the library
+    void* source_library = dlopen(settings::path_source_library.c_str(),RTLD_LAZY);
+    if(!source_library) {
+      std::stringstream msg("Couldnt open source library " + settings::path_source_library);
+      fatal_error(msg);
+    }
+
+    // load the symbol
+    typedef Particle::Bank (*sample_t)();
+
+    // reset errors
+    dlerror();
+
+    // get the function from the library
+    sample_t sample_source = (sample_t) dlsym(source_library, "sample_source");
+    const char *dlsym_error = dlerror();
+    
+    // check for any dlsym errors
+    if (dlsym_error) {
+      std::cout << dlsym_error << std::endl;
+      dlclose(source_library);
+      fatal_error("Couldnt open the sample_source symbol");
+    }
+
+    // Generation source sites from specified distribution in the
+    // library source 
+    for (int64_t i = 0; i < simulation::work_per_rank; ++i) {
+      // initialize random number seed
+      int64_t id = simulation::total_gen*settings::n_particles +
+        simulation::work_index[mpi::rank] + i + 1;
+      set_particle_seed(id);
+
+      // sample external source distribution
+      simulation::source_bank[i] = sample_source();
+    }
+    // release the library
+    dlclose(source_library);
```

Comment:
Similarly, this whole block should probably be within an ifdef. For non-POSIX systems, we should probably raise an exception here.

Reply:
Still need an ifdef on this block